src/compaction.c

```diff
@@ -77,39 +79,64 @@ void *AvgCreateContext() {
     AvgContext *context = (AvgContext *)malloc(sizeof(AvgContext));
     context->cnt = 0;
     context->val = 0;
+    context->overflow = 0;
     return context;
 }
 
 void AvgAddValue(void *contextPtr, double value) {
     AvgContext *context = (AvgContext *)contextPtr;
-    context->val += value;
     context->cnt++;
+
+    // Test for overflow
+    if (unlikely(((context->val < 0.0) == (value < 0.0) &&
+                  (fabs(context->val) > (DBL_MAX - fabs(value)))) ||
+                 context->overflow)) {
+        // calculating: avg(t+1) = t*avg(t)/(t+1) + val/(t+1)
+        long double ld_val = context->val;
+        long double ld_value = value;
+        ld_val /= context->cnt;
+        if (context->overflow) {
+            ld_val *= (long double)(context->cnt - 1);
+        }
+        ld_val += (ld_value / (long double)context->cnt);
+        context->val = ld_val;
+        context->overflow = 1;
+    } else { // No Overflow
+        context->val += value;
+    }
 }
 
 int AvgFinalize(void *contextPtr, double *value) {
     AvgContext *context = (AvgContext *)contextPtr;
     if (context->cnt == 0)
         return TSDB_ERROR;
-    *value = context->val / context->cnt;
+    if (unlikely(context->overflow)) {
+        *value = context->val;
+    } else {
+        *value = context->val / context->cnt;
+    }
     return TSDB_OK;
 }
 
 void AvgReset(void *contextPtr) {
     AvgContext *context = (AvgContext *)contextPtr;
     context->val = 0;
     context->cnt = 0;
+    context->overflow = 0;
 }
 
 void AvgWriteContext(void *contextPtr, RedisModuleIO *io) {
     AvgContext *context = (AvgContext *)contextPtr;
     RedisModule_SaveDouble(io, context->val);
     RedisModule_SaveDouble(io, context->cnt);
+    RedisModule_SaveUnsigned(io, context->overflow);
 }
 
 int AvgReadContext(void *contextPtr, RedisModuleIO *io) {
     AvgContext *context = (AvgContext *)contextPtr;
     context->val = LoadDouble_IOError(io, goto err);
     context->cnt = LoadDouble_IOError(io, goto err);
+    context->overflow = LoadUnsigned_IOError(io, goto err);
```

Comment:
My comment on the write was not 100%, the write will not break the RDB but the read will.
1. we need to increase the RDB version to indicate we changed the format.
2. we need to pass the RDB version to the read context and be compatible with older RDB versions.
3. we have tests to verify compatibility but its probably not covering the aggregations context